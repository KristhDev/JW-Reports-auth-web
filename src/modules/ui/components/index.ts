export { Button } from './Button';
export { EyeBtn } from './EyeBtn';
export { FormField } from './FormField';
export { Spinner } from './Spinner';
export { Title } from './Title';