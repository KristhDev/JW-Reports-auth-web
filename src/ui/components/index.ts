export { FormField } from './FormField';
export { Title } from './Title';