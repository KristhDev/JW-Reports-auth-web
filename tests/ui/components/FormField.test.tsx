import { render, screen, waitFor, fireEvent } from '@testing-library/react';

import { IoPersonOutline } from 'react-icons/io5';
import { Formik } from 'formik';

import { FormField } from '../../../src/ui/components';

const fieldLabel = 'Nombre:'
const fieldName = 'name';
const fieldPlaceholder = 'Ingrese su nombre';
const fieldType = 'text';

describe('Test in <FormField /> component', () => {
    beforeEach(() => {
        render(
            <Formik
                initialValues={{ name: '' }}
                onSubmit={ () => {} }
            >
                { () => (
                    <FormField
                        icon={ <IoPersonOutline size={ 25 } /> }
                        label={ fieldLabel }
                        name={ fieldName }
                        placeholder={ fieldPlaceholder }
                        type={ fieldType }
                    />
                ) }
            </Formik>
        );
    });

    it('should to match snapshot', () => {
        const { container } = render(
            <Formik
                initialValues={{ name: '' }}
                onSubmit={ () => {} }
            >
                { () => (
                    <FormField
                        icon={ <IoPersonOutline size={ 25 } /> }
                        label={ fieldLabel }
                        name={ fieldName }
                        placeholder={ fieldPlaceholder }
                        type={ fieldType }
                    />
                ) }
            </Formik>
        );

        expect(container).toMatchSnapshot();
    });

    it('should render respective props', () => {
        const label = screen.getByTestId<HTMLLabelElement>('form-field-label');
        const input = screen.getByRole<HTMLInputElement>('textbox');

        expect(label.innerHTML).toBe(fieldLabel);
        expect(input.name).toBe(fieldName);
        expect(input.placeholder).toBe(fieldPlaceholder);
        expect(input.type).toBe(fieldType);
    });

    it('should change value of input', () => {
        const input = screen.getByRole<HTMLInputElement>('textbox');
        fireEvent.input(input, { target: { value: 'Test Name' } });

        expect(input.value).toBe('Test Name');
    });
});