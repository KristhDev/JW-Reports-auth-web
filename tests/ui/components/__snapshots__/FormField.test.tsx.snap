// Vitest Snapshot v1

exports[`Test in <FormField /> component > should to match snapshot 1`] = `
<div>
  <div
    class="form-field__container"
  >
    <label
      class="form-field__label undefined"
      data-testid="form-field-label"
      for="name"
    >
      Nombre:
    </label>
    <div
      class="form-field__input-container"
    >
      <input
        class="form-field__input-container__input undefined"
        name="name"
        placeholder="Ingrese su nombre"
        type="text"
        value=""
      />
      <svg
        fill="currentColor"
        height="25"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 512 512"
        width="25"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M344 144c-3.92 52.87-44 96-88 96s-84.15-43.12-88-96c-4-55 35-96 88-96s92 42 88 96z"
          fill="none"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="32"
        />
        <path
          d="M256 304c-87 0-175.3 48-191.64 138.6C62.39 453.52 68.57 464 80 464h352c11.44 0 17.62-10.48 15.65-21.4C431.3 352 343 304 256 304z"
          fill="none"
          stroke-miterlimit="10"
          stroke-width="32"
        />
      </svg>
    </div>
  </div>
</div>
`;
